vim-patch:8.2.{4359,4362,4363,4364}: crash when repeatedly using :retab

seandewar opened the pull request:
Problem:    crash when repeatedly using :retab.
Solution:   Bail out when the line is getting too long.
https://github.com/vim/vim/commit/6e28703a8e41f775f64e442c5d11ce1ff599aa3f

Cherry-pick e_resulting_text_too_long from v8.2.3492; put it in globals.h as
it will eventually be used in other files.

Add a modeline to test_retab.vim

seandewar committed:
vim-patch:8.2.4359: crash when repeatedly using :retab

Problem:    crash when repeatedly using :retab.
Solution:   Bail out when the line is getting too long.
https://github.com/vim/vim/commit/6e28703a8e41f775f64e442c5d11ce1ff599aa3f

Cherry-pick e_resulting_text_too_long from v8.2.3492; put it in globals.h as
it will eventually be used in other files.

Add a modeline to test_retab.vim

seandewar committed:
vim-patch:8.2.4362: :retab may allocate too much memory

Problem:    :retab may allocate too much memory.
Solution:   Bail out when allocating more than MAXCOL bytes.
https://github.com/vim/vim/commit/33f3c5985491032d5bdfc30e722e85d5a0285e64

seandewar committed:
vim-patch:8.2.4363: MS-Windows: running out of memory for a very long line

Problem:    MS-Windows: running out of memory for a very long line.
Solution:   Use a 32 bit value for MAXCOL also when ints are 64 bits.
https://github.com/vim/vim/commit/8e38555ece7d3fe1edc6681ec70fe5586a524862

This still fails Vim's Windows CI, so let's see what happens...

seandewar commented:
Generally if the error is used only in that source file we make it static and stick it near the top; otherwise it goes in globals.h (even for the newfangled errors.h ones).

v8.2.3492 is a bit dodgy as the overflow check it introduces is broken and requires even more patches to fix, which also introduced stuff like `v:sizeoflong` which I'm not too enthusiastic about porting (only really used for tests; unsure if that stuff is really needed).

zeertzjq commented:
<https://github.com/vim/vim/commit/45491660787043ea412719544881db691338d730> passed Vim's CI.

seandewar committed:
vim-patch:8.2.4364: MS-Windows: still running out of memory for a very long line

Problem:    MS-Windows: still running out of memory for a very long line.
Solution:   Check for negative length.
https://github.com/vim/vim/commit/45491660787043ea412719544881db691338d730
